From dac571471edb2e266b8298271f2f473b5360494d Mon Sep 17 00:00:00 2001
From: wangbin <wbsecg1@gmail.com>
Date: Fri, 5 Sep 2025 23:11:42 +0800
Subject: [PATCH 33/35] hls: allow playlist w/o extension

there's no option for `read_probe()`, so let's return a lower score.
fix https://github.com/wang-bin/fvp/issues/218
---
 libavformat/hls.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavformat/hls.c b/libavformat/hls.c
index ddfaa21ef9..90cb3387a6 100644
--- a/libavformat/hls.c
+++ b/libavformat/hls.c
@@ -3153,7 +3153,7 @@ static int hls_probe(const AVProbeData *p)
             !av_match_ext    (p->filename, "m3u8,m3u") &&
              ff_match_url_ext(p->filename, "m3u8,m3u") <= 0) {
             av_log(NULL, AV_LOG_ERROR, "Not detecting m3u8/hls with non standard extension and non standard mime type\n");
-            return 0;
+            return AVPROBE_SCORE_MAX / 2;
         }
         if (mime_x)
             av_log(NULL, AV_LOG_WARNING, "mime type is not rfc8216 compliant\n");
-- 
2.54.0 (Apple Git-157)

